gke/master.tf

```diff
@@ -14,59 +14,10 @@ provider "google" {
   project     = "${var.project_id}"
 }
 
-resource "google_container_cluster" "gcp_kubernetes" {
-    provider           = "google-beta"
-    name               = "${var.cluster_name}"
-    location           = "us-west1-a"
-    project            = "${var.project_id}"
-    initial_node_count = 1
-    min_master_version = "1.12"
-    remove_default_node_pool = true
-    master_auth {
-        username = "${var.ssh_username}"
-        password = "${var.ssh_password}"
-    }
-    cluster_autoscaling {
-        enabled = true
-        resource_limits {
-            resource_type = "cpu"
-            maximum = 16
-        }
-        resource_limits {
-            resource_type = "memory"
-            maximum = 64
-        }
-    }
-    addons_config {
-        http_load_balancing {
-            disabled = false
-        }
-    }
-}
-
-resource "google_container_node_pool" "np_kubernetes" {
-    provider           = "google-beta"
-    name               = "${var.cluster_name}-node-pool-0"
-    location           = "us-west1-a"
-    project            = "${var.project_id}"
-    initial_node_count = 1
-    node_config {
-        preemptible  = true
-        oauth_scopes = [
-          "https://www.googleapis.com/auth/compute",
-          "https://www.googleapis.com/auth/devstorage.read_only",
-          "https://www.googleapis.com/auth/logging.write",
-          "https://www.googleapis.com/auth/monitoring",
-        ]
-        tags = ["afrank"]
-    }
-    autoscaling	{
-        min_node_count = 1
-        max_node_count = 5
-    }
-    cluster            = "${var.cluster_name}"
-    management {
-        auto_upgrade = true
-        auto_repair  = true
-    }
+module "gke" {
+  source       = "git@github.com:mozilla-it/terraform-sre-gke.git"
```

Comment:
Following along this source i noticed https://github.com/mozilla-it/terraform-sre-gke/blob/master/main.tf#L6

```
provider = "google-beta"
```

Hard coding this value means that we need to name our provider `google-beta` as well. We should aim to make this more generic.

Also this https://github.com/mozilla-it/terraform-sre-gke/blob/master/main.tf#L57
```
 tags = ["afrank"]
```
Shouldn't tag it as you or hardcode your name as well :)

Reply:
Lol, you're onto my side hustle, getting amazon kick-backs for tag revenue :D

Reply:
For provider, I don't think it's inappropriate to hardcode that. There are features I use that require the google-beta provider (https://www.terraform.io/docs/providers/google/provider_versions.html#using-the-google-beta-provider) so hardcoding it will give the user a meaningful error message they can easily correct if they're missing that provider, IMO.